src/lib/coda_base/transaction_validator.ml

```diff
@@ -0,0 +1,77 @@
+open Base
+open Signature_lib
+
+module Hashless_ledger = struct
+  type t = {base: Ledger.t; overlay: (Account.key, Account.t) Hashtbl.t}
+
+  type location = Ours of Account.key | Theirs of Ledger.Location.t
+
+  let msg s =
+    s
+    ^ ": somehow we got a location that isn't present in the underlying ledger"
+
+  let get t = function
+    | Ours key -> Hashtbl.find t.overlay key
+    | Theirs loc -> (
+      match Ledger.get t.base loc with
+      | Some a -> (
+        match Hashtbl.find t.overlay (Account.public_key a) with
+        | None -> Some a
```

Comment:
nit: `Option.value ~default` instead of a match here

Reply:
As usual, `Option.value` has the wrong type. This is Rust's `Option::or_else`.